feat:  removeBreakWeaponsCharges instead of removeBeginningWeaponAmmunition

jprzimba opened the pull request:
* removeBreakWeaponsCharges  now works for all weapons that have break chance


# Description
 Update break weapown code to work with all break weapons

## How Has This Been Tested
Always use on my old Crystal Server

Commit:
* removeBreakWeaponsCharges instead of removeBeginningWeaponAmmunition

* removeBreakWeaponsCharges  now works for all weapons that have break chance

Commit:
fix: typo in savePlayerStorage function (#2435)

Commit:
refactor: move tools to data path (#2434)

sebbesiren committed:
fix: two-handed being equipped in right hand on unequip (#2436)

Equipping a two-handed weapon using keybind and then unequipping it placed the two-handed weapon in right hand.

The change makes the right hand not a possible slot for a two-handed weapon which forces it to be placed in backpack similar to other equipment.

Dvineone committed:
fix: added check for type 'two-handed' (#2410)

Warnings when starting the server.

dudantas committed:
fix: skill duplication on equip item (#2439)

Revert send mistake from:
https://github.com/opentibiabr/canary/commit/c6895eb063d564a479e4a42e4c1d15f241f9aa9e

sebbesiren commented:
This doesnt solve the same use case as the feature you removed. 

Your feature: 
- Removes break chance on everything

The other feature
- Removed break chance on specific early game ammo  (so paladins can use exercise dummy)

elsongabriel committed:
fix: hazard mount and achievement (#2431)

jprzimba commented:
> * (so paladins can use exercise dummy)

players still use , if using my code
-------------------
The people involved simply don't want help with the project.
they just want money, doing vakinha and bla bla bla

As the Creator and founder of Crystal server, I would be fully capable of collaborating with the project, but I realize that the people involved only want money.

That's why the project isn't going forward and it's full of bugs..... they're putting the map above the distro.
Where have you seen it? Without a stable server there is no point in having a copy of the global map

sebbesiren commented:
With your change all types are affected. With the previous it was only the beginning ammo types etc. Not same feature imo.

sebbesiren commented:
And i use the existing feature since i still want players to purchase ammo such as diamond arrows.

luan committed:
fix: check if bankable is valid player before emitting metric (#2453)

elsongabriel committed:
fix: global server save and expBoostCount improvements (#2449)

dudantas committed:
fix: sonar bugs related to uninitialized variable (#2462)

• Initialized variables
• Guaranteed that there is no division by 0, avoiding crash

dudantas committed:
fix: two-handed weapons attributes (#2461)

Resolves #2442

Co-authored-by: Lpcordovilll <162099109+lpcordovilll@users.noreply.github.com>

dudantas committed:
fix: wheel bugs (#2417)

Added some missing wheel modifiers.
Fixed some wrong wheel slots.

dudantas committed:
fix: register weapon with chain attribute (#2421)

Log moved to trace and fixed a minor error in the deactivation logic.

lamonato29 committed:
improve: banking NPC behavior for gold withdraw (#2414)

This update significantly improves the interaction with banking NPCs by addressing the issue where NPCs would drop gold on the ground if a player didn't have enough free slots in their backpack or enough carrying capacity. Now, instead of dropping the gold, the NPC will warn the player about the lack of space or capacity, preventing potential loss of gold. This change ensures a smoother and more intuitive banking experience, keeping players' assets secure and enhancing overall gameplay satisfaction. 

Fixes #2290.

Tested scenarios include withdrawing gold with full backpacks and exceeding carrying capacity, with NPCs now providing warnings instead of dropping gold.

dudantas committed:
fix: correct configure weapon wand (#2465)

Fix related from the pr: #1494

Resolves #2464
Resolves #2383

CarlosE-Dev committed:
fix: incorrect pricing for enchant/recharge with silver tokens (#2463)

The NPC Cledwyn, who performs enchant/recharge of various items in exchange for silver tokens is ALWAYS charging 2 silver tokens for each operation. However, some of them should cost 5 silver tokens.

Aerwix committed:
fix: set isBlocking to false for area runes (#2468)

Resolves #2467

Commit:
Update items.xml

Commit:
Merge branch 'main' of https://github.com/jprzimba/canary

Commit:
Update items.xml

Commit:
Merge branch 'main' of https://github.com/jprzimba/canary
